src/averageAges.js

```diff
@@ -38,6 +49,16 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const woman = withChildren
+    ? people
+      .filter((person, i, array) => person.sex === 'f'
+        && array.some(a => a.mother === person.name))
+    : people.filter(person => person.sex === 'f');
```

Comment:
Same here, instead of writing two filter methods for both cases, just write one, and select needed condition by ternary operator